src/Microsoft.TemplateEngine.Cli/TemplateResolution/TemplateResolutionResult.cs

```diff
@@ -202,8 +204,52 @@ internal TemplateGroup UnambiguousTemplateGroup
             }
         }
 
+        internal IEnumerable<ITemplateMatchInfo> TemplatesForDetailedHelp
+        {
+            get
+            {
+                if (UnambiguousTemplateGroup == null || !UnambiguousTemplateGroup.InvokableTemplates.Any())
+                {
+                    return Array.Empty<ITemplateMatchInfo>();
+                }
+
+                if (_hasUserInputLanguage)
+                {
+                    return UnambiguousTemplateGroup.InvokableTemplates.Where(t => t.HasLanguageMatch());
+                }
+                else if (UnambiguousTemplateGroup.InvokableTemplates.Any(t => t.HasDefaultLanguageMatch()))
+                {
+                    return UnambiguousTemplateGroup.InvokableTemplates.Where(t => t.HasDefaultLanguageMatch());
```

Comment:
So, that's an interesting part.
When I started this refactoring I thought that `--help` is closer to instantiate that to `--list` as we can show the help only for single template, however the truth in the middle:
- instantiate needs to resolve single template from all the groups / otherwise we cannot proceed
- `--help` doesn't absolutely need single template, however needs single template group and (!) single language. Example of use case when it's not a single template is console which is spread between different templates for different tfms
- `--list` should support multiple template groups, and multiple templates in them.

The logic for help is as following:
- it needs single group to be resolved
- language logic 
  - if user specifies the language, that language is selected
  - if not - default language is attempted
  - if not - we check if all the templates inside group have same language, then we still can show the help
  - if not, we fall back to  `CoordinateAmbiguousTemplateResolutionDisplayAsync` error handling

Previously `CoordinateAmbiguousTemplateResolutionDisplayAsync` was only used for instantiate, however I find out that it also works for help as use cases are very similar. 

I still have a plan to do major refactoring of this part, once we cut all features that we don't need. Rough outline:
- we start with creating the template groups. this allows to do extra features, for example short name sharing
- we filter template groups based on user input first, then we filter individual templates inside the groups. Potentially this will need new class, similarly to `ITemplateInfo` - `ITemplateMatchInfo`, which will now the context of the command specified by user. This class will handle the matches and different getters (i.e. get templates matching default language, get invokable templates (i.e. templates matching all params), get single invokable template, get templates that match exactly, partially etc).
- `TemplateResolver` probably becomes non-static with base class and different implementations for instantiate, help, list and search - we need different sets of filter and different exit criteria. Or alternatively handlers will decide it `TemplateResolutionResult` fits their need or not.  
- `TemplateResolutionResult` - at the moment I'm thinking to do single implementation and merge *List* features into it as they also useful for other cases. As we discussed before, result class won't have any logic in, just static representation of result. All logic will go to `TemplateResolver`. 

I split list and help handlers as in couple of months, they will probably evolve to corresponding subcommands handlers. Having mixed logic for help and list doesn't make sense.